pkgs/applications/misc/st/default.nix

```diff
@@ -4,22 +4,20 @@
 with stdenv.lib;
 
 stdenv.mkDerivation rec {
-  name = "st-0.6";
+  name = "st";
```

Comment:
It is important that the version is part of the name. E.g, it should be `name = "st-0.7";` for this updated version. See the [Package naming](https://nixos.org/nixpkgs/manual/#sec-package-naming) section of the Nixpkgs manual.